src/Microsoft.AspNet.Mvc.Razor/Compilation/RazorCompilationService.cs

```diff
@@ -16,12 +19,15 @@ public class RazorCompilationService : IRazorCompilationService
     {
         private readonly ICompilationService _compilationService;
         private readonly IMvcRazorHost _razorHost;
+        private readonly IFileProvider _fileProvider;
 
         public RazorCompilationService(ICompilationService compilationService,
-                                       IMvcRazorHost razorHost)
+                                       IMvcRazorHost razorHost,
+                                       IOptions<RazorViewEngineOptions> viewEngineOptions)
```

Comment:
The file system that we use for views specifically hangs off of `RazorViewEngineOptions`. It allows doing things like using resources as view sources - https://github.com/aspnet/Mvc/blob/dev/test/WebSites/RazorEmbeddedViewsWebSite/Startup.cs#L20-L23